config/hive-site-template.xml

```diff
@@ -0,0 +1,47 @@
+<configuration>
+  <property>
+    <name>javax.jdo.option.ConnectionURL</name>
+    <value>jdbc:postgresql://{{POSTGRES_URL}}/{{POSTGRES_DB}}</value>
+  </property>
+
+  <!-- JDBC driver class name for PostgreSQL -->
+  <!--  Ensure that PostgreSQL JDBC driver jars are included via Gradle -->
+  <property>
+    <name>javax.jdo.option.ConnectionDriverName</name>
+    <value>org.postgresql.Driver</value>
+  </property>
+  <property>
+    <name>javax.jdo.option.ConnectionUserName</name>
+    <value>{{POSTGRES_USER}}</value>
+  </property>
+  <property>
+    <name>javax.jdo.option.ConnectionPassword</name>
+    <value>{{POSTGRES_PASSWORD}}</value>
+  </property>
+
+  <!-- Configuration to automatically create the necessary tables in the database schema if they don't exist -->
+  <!-- Hive metastore fails to start if this property is not set-->
+  <property>
+    <name>datanucleus.schema.autoCreateTables</name>
+    <value>true</value>
```

Comment:
This means the notebook container has to have create table perms on the postgres instance. It seems like it'd be better if the tables were created beforehand and the container only has write to those tables

Reply:
I don't think any of us fully understand which tables to create. It's probably safer to let it create what it needs.

Reply:
Can we run a notebook with create table privs, have hive create the tables, and then restart the notebook with just write? Or have a separate notebook with just write?

Reply:
I guess yea we can introduce a env variable to control this behavior. But what if a table needs to be created in the middle of operations? I guess if we are creating a DB separately just for hive, would that be fine? What’s bad about giving hive the privilege to create what it needs?

Reply:
If hive can create tables in the db, then so can anyone using the notebook container. Not sure what other privs the creds we'll give to the container provide as well.

Basically the same idea as giving reduced creds for minio to reduce the chance of things getting borked up

Reply:
👍 added `USAGE_MODE` to toggle settings accordingly.

Reply:
:+1: